ScreenBoom/ScreenBoom/ViewController/HomeViewController.swift

```diff
@@ -12,39 +12,152 @@ import FirebaseDatabase
+    return view
+  }()
+  
+  func animationController(forPresented presented: UIViewController, presenting: UIViewController, source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    transition.transitionMode = .present
+    transition.startingPoint = createEventBottun.center
+    transition.circleColor = createEventBottun.backgroundColor!
+    return transition
+  }
+  
+  func animationController(forDismissed dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+    transition.transitionMode = .dismiss
+    transition.startingPoint = createEventBottun.center
+    transition.circleColor = createEventBottun.backgroundColor!
+    return transition
+  }
   
   override func viewDidLoad() {
     super.viewDidLoad()
-//    firebaseDatabaseReference = Database.database().reference()
-    if !isDeepLinking {
-      rigisterUser()
-    }
+    
+    
+    setupViews()
+    
+    rigisterUser()
+//    if !isDeepLinking { rigisterUser() }
   }
   
-  override func viewWillLayoutSubviews() {
-    let isLandscape = UIDevice.current.orientation.isLandscape
-    if isLandscape {
-      topDistance.constant = Constant.imageTopDistanceLandScape
-      return
-    }
-    topDistance.constant = Constant.imageTopDistancePortrait
+  
+  
+  override func viewWillAppear(_ animated: Bool) {
+    self.navigationController?.isNavigationBarHidden = true
+  }
+  
+  private func setupViews() {
+    
+    let topImageContainerView : UIView = {
+      let view = UIView()
+      return view
+    }()
+    
+    view.addSubview(topImageContainerView)
+    view.addSubview(bottunStackView)
+    view.addSubview(desceptionTextView)
+    topImageContainerView.addSubview(logoImageView)
+    
+    topImageContainerView.anchor(top: view.topAnchor,
+                                 leading: view.leadingAnchor,
+                                 bottom: nil,
+                                 trailing: view.trailingAnchor)
+    topImageContainerView.heightAnchor.constraint(equalTo: view.heightAnchor, multiplier: 0.5).isActive = true
+    
+    NSLayoutConstraint.activate([
+      logoImageView.centerXAnchor.constraint(equalTo: topImageContainerView.centerXAnchor),
+      logoImageView.centerYAnchor.constraint(equalTo: topImageContainerView.centerYAnchor),
+      logoImageView.heightAnchor.constraint(equalTo: topImageContainerView.heightAnchor, multiplier: 0.5),
+      logoImageView.widthAnchor.constraint(equalTo: logoImageView.heightAnchor, multiplier: 4/3)
+      ])
+    
+    bottunStackView.addArrangedSubview(createEventBottun)
+    bottunStackView.addArrangedSubview(joinEventBottun)
+    bottunStackView.anchor(top: topImageContainerView.bottomAnchor,
+                           leading: view.safeAreaLayoutGuide.leadingAnchor,
+                           bottom: nil,
+                           trailing: view.safeAreaLayoutGuide.trailingAnchor,
+                           padding: .init(top: 10, left: 10, bottom: 0, right: 10),
```

Comment:
like I said before about the random numbers. Please put them in a struct as static constants